specs/Menu.md

```diff
@@ -1,731 +1,672 @@
+
+Sample usages will be give in the following section of this document [Sample code](#sample-code)
+
+### Menu
+
+The root level component serves as a simplified interface (sugar) for popup positioning and triggering.
+
+### MenuList
+
+This component is used internally by `Menu` and manages the context and layout its items.
+
+`MenuList` can also be used separately as the standalone variant of the `Menu`, since it should not control popup positioning or triggers. It is the only component in the API that can be used standalone. Envisioned to be used with more complex popup or trigger scenarios where the `Menu` component does not provide enough control for these situations.
+
+### MenuSection
+
+Creates a section inside a `MenuList`, setting up header layout and dividers between `MenuItems`.
 
-### Slots
+The MenuSection is also a useful component to declare different selection groups (checkbox/radio) in a `MenuList`.
 
-#### Menu
+### MenuDivider
 
-Since the Menu is only rendering its children, the only slot necessary is the one for the Root `div`. The rest can be passed in as children variants.
+Creates a divider element in the `MenuList` with correct HTML and aria semantics for divider. Intention is to use this internally within `MenuList` and `MenuSection` and avoid explicit uses of dividers to ensure correct HTML and aria semantics.
 
-#### MenuItem
+### MenuItem
 
-Since the MenuItem is only rendering its children, the only slot necessary is the one for the Root `div`. The rest can be passed in as children.
+As the name infers
 
-#### SubmenuList
+### SubMenu
 
-Since the SubmenuList is effectively the same as the Menu, it should have only one slot for the Root `div` and for all intents and purposes should behave exactly the same as the Menu slot does.
+Creates a `Menu` component with `MenuItem` trigger and handles the positioning of the nested menu.
 
-#### SubmenuItem
+## Sample code
 
-Since the SubmenuItem is effectively the same as the MenuItem, it should have only one slot for the Root `div` and for all intents and purposes should behave exactly the same as the MenuItem slot does.
+The below samples do not represent the definitive props of the final implemented component, but represent the ideal final implementations. Can be subject to change during the implementation phase.
 
-### Behaviors
+### Default Menu
 
-#### Menu
+```typescript
+const trigger = <button> Open menu </button>
 
-The menu itself should be a list that renders menu items, it will provide context about its overall state but should not pass that directly into its items as props.
+const menu = (
+  <Menu trigger={trigger}>
+    <MenuItem>Option 1</MenuItem>
+    <MenuItem>Option 2</MenuItem>
+    <MenuItem>Option 3</MenuItem>
+  <Menu>
+)
+```
 
-##### Orientation
+```html
+<!-- expected DOM output  -->
+<!-- TODO positioning -->
+<div role="menu">
+  <div role="menuitem" tabindex="0">Option 1</div>
```

Comment:
Have you verified a11y is ok with just `div`s?

Reply:
Yes thanks to @kolaps33